docs/specs/2fa-grace-period.md

```diff
@@ -0,0 +1,81 @@
+# 2FA Grace Period
+
+- [Abstract](#abstract)
+- [Configuration](#configuration)
+  - [Global Grace Period](#global-grace-period)
+  - [Per-user Grace Period](#per-user-grace-period)
+  - [Customized Authentication Flow](#customized-authentication-flow)
+
+## Abstract
+
+In order to enforce 2FA, we will provide a grace period for users to enroll in 2FA after they have signed up.
+
+After the grace period, the user will be required to enroll in 2FA before they can sign in.
+
+## Configuration
+
+By default, no grace period is provided. User must enroll in 2FA immediately after [2FA mode](./user-model.md#secondary-authenticator) is set to `required`.
```

Comment:
Change default behaviour to force enroll or add another mode? 

Auth0 always allows user to enroll on login https://auth0.com/docs/secure/multi-factor-authentication/customize-mfa/customize-mfa-enrollments-universal-login#how-it-works

Reply:
Default behaviour will remain block instead of force enrollment after grace period.